metrics-datadog/src/main/java/org/coursera/metrics/datadog/transport/HttpTransport.java

```diff
@@ -116,20 +128,26 @@ public void send() throws Exception {
         StringBuilder sb = new StringBuilder();
         sb.append("Sending HTTP POST request to ");
         sb.append(this.transport.seriesUrl);
-        sb.append(", POST body length is: ");
+        sb.append(", uncompressed POST body length is: ");
         sb.append(postBody.length());
         LOG.debug(sb.toString());
 
         StringBuilder bodyMsgBuilder = new StringBuilder();
-        bodyMsgBuilder.append("POST body is: \n").append(postBody);
+        bodyMsgBuilder.append("Uncompressed POST body is: \n").append(postBody);
         LOG.debug(bodyMsgBuilder.toString());
       }
       long start = System.currentTimeMillis();
       org.apache.http.client.fluent.Request request = Post(this.transport.seriesUrl)
         .useExpectContinue()
         .connectTimeout(this.transport.connectTimeout)
-        .socketTimeout(this.transport.socketTimeout)
-        .bodyString(postBody, ContentType.APPLICATION_JSON);
+        .socketTimeout(this.transport.socketTimeout);
+        if (this.transport.useCompression) {
+          request.addHeader("Content-Encoding", "deflate")
```

Comment:
Style nit: Put the first `.addHeader` starting on its own line, aligned with other chained method calls.